src/06-excitation-transfer-revisited.md

```diff
@@ -0,0 +1,696 @@
+        title=f"Even parity energy levels for \n {H_latex}  \n $\omega={omega}$, $\Delta E=2.5$   (Fig. 4)",
+             figsize=(10,8));
+
+plt.ylabel("Energy");
+
+```
+
+```python
+df_even23 = df_even[["$U$", "level_2", "level_3"]] # Only keep levels 2 and 3
+```
+
+```python
+# Calculate the excitation transfer time
+df_even23["Time"] = 2*np.pi / (df_even23["level_3"] - df_even23["level_2"]) 
+```
+
+```python
+df_even23.plot(x="$U$",y="Time", title=f"Excitation transfer time for matched TLS  \n {H_latex} \n $\omega={omega}$, $\Delta E=2.5$   (Fig. 5)", legend=False);
+plt.ylabel("T"); 
+```
+
+Much like Fig. 2,  Fig. 5 also looks polynomial in nature, but with a negative power this time. Let's verify this by viewing it in as a log-log plot.
+
+```python
+df_even23.plot(x="$U$",y="Time",logx = True, logy = True, title=f"Excitation transfer time for matched TLS  \n {H_latex} \n $\omega={omega}$, $\Delta E=2.5$   (Fig. 6)", legend=False);
+plt.xlabel("$\log(U)$")
+plt.ylabel("$\log(T)$");
+```
+
+A nice straight line downwards confirms our suspicion. Now let's check out the slope:
+
+```python
+linregress(np.log10(df_even23["$U$"]), np.log10(df_even23["Time"])).slope
+```
+
+We've got an inverse quadratic relationship between coupling strength and excitation transfer time $T \sim U ^{-2}$.
+
+When we combine the two scalings we reproduce the 2008 analytical findings from [Hagelstein](http://dx.doi.org/10.1088/0953-4075/41/13/135501) that excitation transfer time scales like:
+
+$$T \sim \left(\frac{\Delta E}{U}\right)^2$$
+
+when the two TLS are matched.
+
+What happens when the systems are not matched?
+
+
+## 6.3 - Mismatched TLS
+
+
+Let's start by giving the TLS different transition energies but we'll keep the coupling the same for each. In other words:
+
+$$H =  \frac{\Delta E_1}{2} \sigma_{z1} + \frac{\Delta E_2}{2}\sigma_{z2} + \hbar\omega\left(a^{\dagger}a +\frac{1}{2}\right) + U\left( a^{\dagger} + a \right)(\sigma_{x1} + \sigma_{x2})$$
+
+
+```python
+H_latex = "$H = (\Delta E_1 /2) \sigma_{z1} + (\Delta E_2 /2) \sigma_{z2} + \hbar\omega(a^{{\dagger}}a +1/2) + U( a^{{\dagger}} + a )(\sigma_{x1} +\sigma_{x2} )$ "
+```
+
+Let's again work with $U=0.1$
+
+```python
+U=0.1
+```
+
+We'll fix $\Delta E_2 = 2.5$ and scan through $\Delta E_1$ to look at the energy level landscape
+
+```python
+DeltaE2 = 2.5
+```
+
+```python
+# prepare data structure for the energy level scan
+df_even = make_df_for_energy_scan("$\Delta E_1$", 0, 200, 2001, two_state_1.shape[0])
+
+for i, row in df_even.iterrows():
+    H =  row["$\Delta E_1$"]*two_state_1+ DeltaE2*two_state_2 + omega*bosons + U*interaction_1 + U*interaction_2
+    evals, ekets = H.eigenstates()
+    df_even.iloc[i,1:] = evals 
+```
+
+```python
+df_even.plot(x="$\Delta E_1$",ylim=[0,6],xlim=[0,9], legend=True, 
+        title=f"Even parity energy levels for \n {H_latex} \n  $\Delta E_2={DeltaE2}$ $\omega={omega}$, $U={U}$   (Fig. 7)",
+             figsize=(10,8));
+
+plt.ylabel("Energy");
```

Comment:
This is the total energy in the Hamiltonian, right? Could be good to make this more clear. Maybe call it "total energy"?

Reply:
What we are plotting is the eigenvalues of the hamiltonian that's written in the figure title. The hamiltonian is the total energy, so I think we're covered. But perhaps I'm missing something. What other energy would you imagine it could be?

Reply:
That's great. Shall we maybe add that sentence that you just wrote here to the main text, where this kind of graph is first introduced?